src/app/filecoin-plus/allocators/utils/getAllocatorsWithDatacap.ts

```diff
@@ -0,0 +1,27 @@
+import prettyBytes from 'pretty-bytes'
+
+import { getAllocators } from '../utils/getAllocators'
+import { getDatacapStats } from '../utils/getDatacapStats'
+
+export async function getAllocatorsWithDatacap() {
+  const allocators = await getAllocators()
+  const datacapStats = await getDatacapStats()
+
+  const allocatorsWithDatacap = allocators.map((allocator) => {
+    const stats = datacapStats.find(
+      (stats) => stats.address === allocator.address,
+    )
+
+    return {
+      ...allocator,
+      remainingDatacap: formatDatacap(stats?.remainingDatacap),
+      allowance: formatDatacap(stats?.allowance),
+    }
+  })
+
+  return allocatorsWithDatacap
+}
+
+function formatDatacap(value?: string) {
+  return value ? prettyBytes(Number(value)) : 'No data available'
+}
```

Comment:
If feel like this formatting/rendering logic should go in `allocatorsTableColumnsData`. `getAllocatorsWithDatacap` should only be concerned about returning the data.

Something like

```ts
cell: (info) => info.getValue() || 'No data available'
```

What do you think?

Reply:
I like that!

Reply:
Actually on this one, we need to provide fallback here 

```
    return {
      ...allocator,
      remainingDatacap: stats?.remainingDatacap ?? 'No data available',
      allowance: stats?.allowance ?? 'No data available',
    }
```
Wondering if it would make sense to leave the format function rather than provide an empty string or 'No data available' - any thoughts?